### DISCUSSION TITLE
Pactumjs with graphql and mochawesome reporter

### DISCUSSION THREAD

**Discussion by userjoshs** [pactumjs/pactum#103] [Q&A] [answered]
Hi all,
I am looking for a js library to implement an api testing framework for my project and came across this library.
Can someone please help me find answers to following questions:
 1.  can I use pactum js with graphql 
 2.  To generate report can it be integrated with mochawesome reporter as we are already using mochawesome  in our project?
 3. Can I set proxies while using pactum?
 4.  Is contract testing possible using Pactum js?
 Thanks in advance.

**Comment by ASaiAnudeep** [comment, marked as answer]
Hey @userjoshs, welcome to **PactumJS** discussions.

> 1. can I use pactum js with graphql

Yes. Please find the docs [here](https://pactumjs.github.io/#/request-making?id=graphql). The library has dedicated methods to send GraphQL requests.

> 2. Integrate with mochawesome reporter

Yes, we can use any kind of reporter that is supported by the respective test runners.

> 3. Can I set proxies while using pactum?

Currently proxies are not supported in pactum's mock server. We have this feature in our roadmap. Can you please give us more details with examples on this proxies to help us better understand your requirement.

> 4. Is contract testing possible using Pactum js?

Yes, contract testing is supported for normal REST calls. We are planning to extend it to GraphQL as well.
